app/models/related_records.rb

```diff
@@ -76,45 +52,27 @@ def clean_id(id)
 
   private
 
-  def fetch_count(id)
-    Rails.cache.fetch("related_records_count/#{id}", expires_in: 15.minutes) do
-      search_service = Blacklight.repository_class.new(blacklight_config)
-      response = search_service.search(
-        q: "parent_id_ssim:\"#{clean_id(id)}\"",
-        rows: 0
-      )
-      if response.present? && response["response"].present?
-        response["response"]["numFound"]
-      else
-        0
+  def derive_collection_name
```

Comment:
Oh, I'll have a look. Thanks!

Reply:
@scott-yeadon Have gone ahead and refactored the code to look for the titles in the Solr document, rather than trying to parse it from the MARC record.